go.mod

```diff
@@ -6,8 +6,7 @@ require (
 	cosmossdk.io/errors v1.0.1
 	cosmossdk.io/math v1.3.0
 	github.com/celestiaorg/blobstream-contracts/v3 v3.1.0
-	github.com/celestiaorg/go-square v1.1.0
-	github.com/celestiaorg/go-square/merkle v0.0.0-20240117232118-fd78256df076
+	github.com/celestiaorg/go-square/v2 v2.0.0-rc2
```

Comment:
[question] should we cut an official release of go-square and bump to that here?

Reply:
Once this is merged confirming that no other changes are necessary I will cut the final release